### ISSUE TITLE
Batch_size affects (changes) results on MMLU

### ISSUE THREAD

**Issue by optas** [oumi-ai/oumi#104]
By changing the batch-size (e.g., 2 vs. 4) we get different accuracy results when using subsets of the MMLU dataset.

This is happening either with the hardcoded dataset subset-selection code: 
https://github.com/openlema/lema/blob/main/src/lema/datasets/mmlu.py#L194

and, with the newer code which allows you to subsample at random, questions of the MMLU dataset.

Note:
  1. Adding `from transformers import set_seed; and set_seed(random_seed)` does not fix the issue.
  2. Both models which exhibited the behavior (our) Zephyr, and a pre-trained Phi3: were loaded/used correctly in "eval.mode".
  3. There is this reference, stating that changing the batch-size changed their evaluation results: https://huggingface.co/spaces/open-llm-leaderboard/open_llm_leaderboard/discussions/82#64b190602fc8324fcb0d4cdf
